### PR TITLE
Added a keymap for kinesis advantage.

### PR DESCRIPTION
**Pull request by heatxsink**
Supported layouts:
- Qwerty
- Colemak Mod-DH
- Workman
- Keypad (Number pad & Media keys)

### CODE REVIEW

File: keyboards/kinesis/keymaps/heatxsink/keymap.c [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+enum layer_names {
+    _DEFAULT = SAFE_RANGE,
```
Comment (drashna) [inline keyboards/kinesis/keymaps/heatxsink/keymap.c:20]: You do not want safe range here.  In fact, that value will probably cause this to outright error during compile 
```suggestion
    _DEFAULT,
```

File: keyboards/kinesis/keymaps/heatxsink/rules.mk [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+MIDI_ENABLE = no            # MIDI controls
+AUDIO_ENABLE = no           # Audio output on port C6
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight.
+SLEEP_LED_ENABLE = yes      # Breathing sleep LED during USB suspend
```
Comment (drashna) [inline keyboards/kinesis/keymaps/heatxsink/rules.mk:13]:
```suggestion
NKRO_ENABLE = yes           # Enable N-Key Rollover
SLEEP_LED_ENABLE = yes      # Breathing sleep LED during USB suspend
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by heatxsink** [comment]
Done.

File: keyboards/kinesis/keymaps/heatxsink/keymap.c [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+            writePinLow(LED_CAPS_LOCK_PIN);
+            writePinLow(LED_COMPOSE_PIN);
+            break;
+        default:
+            // Default layer
+            writePinHigh(LED_NUM_LOCK_PIN);
+            writePinHigh(LED_SCROLL_LOCK_PIN);
+            writePinHigh(LED_COMPOSE_PIN);
+            writePinHigh(LED_CAPS_LOCK_PIN);
+            break;
```
Comment (drashna) [inline keyboards/kinesis/keymaps/heatxsink/keymap.c:235]: From a simplicity standpoint, you could write high to start with, and then write low only for the pins that need to be.  

```suggestion
    writePinHigh(LED_NUM_LOCK_PIN);
    writePinHigh(LED_SCROLL_LOCK_PIN);
    writePinHigh(LED_COMPOSE_PIN);
    writePinHigh(LED_CAPS_LOCK_PIN);
    switch (get_highest_layer(state)) {
        case _QWERTY:
            // Caps Lock Lit
            writePinLow(LED_CAPS_LOCK_PIN);
            break;
        case _COLEMAK_MOD_DH:
            // Num Lock Lit
            writePinLow(LED_NUM_LOCK_PIN);
            break;
        case _WORKMAN:
            // Scroll Lock Lit
            writePinLow(LED_SCROLL_LOCK_PIN);
            break;
        case _KEYPAD:
            // Compose Lit
            writePinLow(LED_COMPOSE_PIN);
            break;
        case _LAYERS:
            // All LEDs Lit
            writePinLow(LED_SCROLL_LOCK_PIN);
            writePinLow(LED_NUM_LOCK_PIN);
            writePinLow(LED_CAPS_LOCK_PIN);
            writePinLow(LED_COMPOSE_PIN);
            break;
```

This isn't required, but may be a bit cleaner.

File: keyboards/kinesis/keymaps/heatxsink/keymap.c [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+}
+
+bool led_update_user(led_t led_state) {
+    // disable led_update_kb() so that we do not override layer indicators
+    return false;
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    return true;
+}
```
Comment (drashna) [inline keyboards/kinesis/keymaps/heatxsink/keymap.c:247]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/kinesis/keymaps/heatxsink/keymap.c (hunk 3 above)
Comment (heatxsink) [reply keyboards/kinesis/keymaps/heatxsink/keymap.c:235]: Thank you for pointing this out ... I dig cleaner!

**Comment by heatxsink** [comment]
Done.

**Review by drashna — APPROVED** [review]

File: keyboards/kinesis/keymaps/heatxsink/rules.mk [hunk 5]
```diff
@@ -0,0 +1,5 @@
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+SLEEP_LED_ENABLE = yes      # Breathing sleep LED during USB suspend
```
Comment (fauxpark) [inline keyboards/kinesis/keymaps/heatxsink/rules.mk:5]:
```suggestion
BOOTMAGIC_ENABLE = no
MOUSEKEY_ENABLE = no
EXTRAKEY_ENABLE = yes
NKRO_ENABLE = yes
SLEEP_LED_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by heatxsink** [comment]
thank you!

**Pull request merged**